src/iterator_trait.rs

```diff
@@ -0,0 +1,15 @@
+use crate::value::Value;
+use bytes::Bytes;
+use enum_dispatch::enum_dispatch;
+
+/// `AgateIterator` defines the interface of all iterators,
+/// including `TableIterator`, `MergeIterator` and `ConcatIterator`.
+#[enum_dispatch]
+pub trait AgateIterator {
```

Comment:
how about `AgateIterator` -> `Iterator` and `Iterator` -> `TableIterator`?

Reply:
Seems that I would confuse this `Iterator` with Rust `Iterator`. What about `TableIterator` and `AgateIterator`?

Reply:
okay

Reply:
`Iterator` -> `TableRefIterator`, and `TableRefIterator<Arc<Table>>` is now `TableIterator`